app/src/main/java/com/penelope/faunafinder/presentation/elements/TextElement.java

```diff
@@ -0,0 +1,143 @@
+package com.penelope.faunafinder.presentation.elements;
+
+import android.graphics.Typeface;
+import android.graphics.text.LineBreaker;
+import android.os.Build;
+import android.util.TypedValue;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import androidx.core.content.res.ResourcesCompat;
+
+import com.penelope.faunafinder.R;
+import com.penelope.faunafinder.xml.slide.Slide;
+
+
+
+public class TextElement extends PresentationElement implements ViewElement{
+    private String content;
+    private int fontSize;
+    private int width;
+    private int height;
+    private long timeOnScreen;
+    private int color;
+    private String font;
+
+    public TextElement(String font,int fontSize, int color,
+                           int x, int y,int width, int height,
+                           long timeOnScreen) {
+        super(x, y);
+       create(font,fontSize,color,width,height,timeOnScreen);
+    }
+
+    //Inherited methods
+    @Override
+    public String getViewType() {
+        return TEXT_ELEMENT;
+    }
+    @Override
+    public String getSearchableContent() {
+        return content.toLowerCase();
+    }
+    @Override
+    public View applyView(View parent, ViewGroup container, Slide slide, int id) {
+        TextView textView = (TextView) parent.findViewById(id);
+        ViewGroup.MarginLayoutParams mlp= (ViewGroup.MarginLayoutParams) textView.getLayoutParams();
+        setTextViewWHParams(slide,mlp);
+        setTextViewXYParams(slide,mlp);
+        setTextViewFontParams(parent, textView);
+        setTextViewTimer(textView);
+        return textView;
+    }
+    //Private methods
+    private void setTextViewWHParams(Slide slide, ViewGroup.MarginLayoutParams mlp) {
+        if (width > 0) {
+            mlp.width = Math.round((width * slide.getCalculatedWidth()) / (float) slide.getWidth());
+        } else if (width == -1) {
+            mlp.width = RelativeLayout.LayoutParams.MATCH_PARENT;
+        } else if (width == -2) {
+            mlp.width = RelativeLayout.LayoutParams.WRAP_CONTENT;
+        } else {
+            mlp.width = 0; // May need to ask Guiseppe what happens when width/height =! >0 , -1 or -2
+        }
+        if (height > 0) {
+            mlp.height = dpToPx(height);
+        } else if (height == -1) {
+            mlp.height = RelativeLayout.LayoutParams.MATCH_PARENT;
+        } else if (height == -2) {
+            mlp.height = RelativeLayout.LayoutParams.WRAP_CONTENT;
+        } else {
+            mlp.height = 0;
+        }
+    }
+    private void setTextViewXYParams(Slide slide,ViewGroup.MarginLayoutParams mlp ) {
+        int leftMarg = Math.round((x * slide.getCalculatedWidth()) / (float) slide.getWidth());
+        int rightMarg = leftMarg + mlp.width;
```

Comment:
This breaks functionality. No text view is displayed. Why is the right margin = the left margin + the width of the component? x,y are top left of components as generally agreed in standard discussion